Services/data/devicemodels/scripts/SetFuelLevel-method.js

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+/*global log*/
+/*global updateState*/
+/*global updateProperty*/
+/*global sleep*/
+/*jslint node: true*/
+
+"use strict";
+
+/**
+ * Entry point function called by the simulation engine.
+ *
+ * @param context        The context contains current time, device model and id
+ * @param previousState  The device state since the last iteration
+ * @param previousProperties  The device properties since the last iteration
+ */
+
+/*jslint unparam: true*/
+function main(context, previousState, previousProperties) {
+
+    // General info indicating that this method is being called
+    log("Executing JavaScript SetFuelLevel method.");
+
+    // Input validation. Make sure that thew new
+    // fuel level is within the allowable range
+    var newFuelLevel = context.FuelLevel;
+    if (isNaN(newFuelLevel)) {
```

Comment:
should you throw or exit instead of setting it to zero if the param isn't passed?

Reply:
Good catch Jill. I'm going to look at implementing an error callback so that the caller will be informed that the method call didn't succeed.

Reply:
I found a way to throw an exception in the JavaScript code which will be brought back up to the CLR. If the input is invalid there will be an error logged.